x/pool-incentives/keeper/distr.go

```diff
@@ -0,0 +1,175 @@
+func (k Keeper) GetAllocatableAsset(ctx sdk.Context) sdk.Coin {
+	params := k.GetParams(ctx)
+
+	feeCollector := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName)
+	asset := k.bankKeeper.GetBalance(ctx, feeCollector.GetAddress(), params.MintedDenom)
+
+	return sdk.NewCoin(asset.Denom, asset.Amount.ToDec().Mul(params.AllocationRatio).TruncateInt())
+}
+
+// AllocateAsset allocates and distributes coin according a pot’s proportional weight that is recorded in the record
+func (k Keeper) AllocateAsset(ctx sdk.Context, asset sdk.Coin) error {
+	distrInfo := k.GetDistrInfo(ctx)
+
+	if distrInfo.TotalWeight.GT(sdk.ZeroInt()) {
+		assetAmountDec := asset.Amount.ToDec()
+		totalWeightDec := distrInfo.TotalWeight.ToDec()
+		for _, record := range distrInfo.Records {
+			allocatingAmount := assetAmountDec.Mul(record.Weight.ToDec().Quo(totalWeightDec)).TruncateInt()
+			coins := sdk.NewCoins(sdk.NewCoin(asset.Denom, allocatingAmount))
+
+			err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, k.feeCollectorName, types.ModuleName, coins)
+			if err != nil {
+				return err
+			}
+
+			err = k.incentivesKeeper.AddToPotRewards(ctx, k.accountKeeper.GetModuleAddress(types.ModuleName), coins, record.PotId)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		// If there are no records, put the asset to the community pool
+		err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, k.feeCollectorName, k.communityPoolName, sdk.Coins{asset})
+		if err != nil {
+			return err
+		}
+
+		feePool := k.distrKeeper.GetFeePool(ctx)
+		feePool.CommunityPool = feePool.CommunityPool.Add(sdk.NewDecCoinsFromCoins(asset)...)
+		k.distrKeeper.SetFeePool(ctx, feePool)
+	}
+
+	return nil
+}
+
+func (k Keeper) GetDistrInfo(ctx sdk.Context) types.DistrInfo {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.DistrInfoKey)
+
+	if len(bz) == 0 {
+		distrInfo := types.DistrInfo{
+			TotalWeight: sdk.NewInt(0),
+			Records:     nil,
+		}
+		bz = k.cdc.MustMarshalBinaryBare(&distrInfo)
+
+		store.Set(types.DistrInfoKey, bz)
+		return distrInfo
+	}
+
+	distrInfo := types.DistrInfo{}
+	k.cdc.MustUnmarshalBinaryBare(bz, &distrInfo)
+
+	return distrInfo
+}
+
+func (k Keeper) SetDistrInfo(ctx sdk.Context, distrInfo types.DistrInfo) {
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshalBinaryBare(&distrInfo)
+	store.Set(types.DistrInfoKey, bz)
+}
+
+// indexOfDistrRecordByPotId returns the index of the record for the specific pot id.
+// If there is no record matched to the pot id, return -1.
+func (k Keeper) indexOfDistrRecordByPotId(ctx sdk.Context, potId uint64) int {
+	distrInfo := k.GetDistrInfo(ctx)
+	records := distrInfo.Records
+	for i, record := range records {
+		if record.PotId == potId {
+			return i
+		}
+	}
+	return -1
+}
+
+func (k Keeper) AddDistrRecords(ctx sdk.Context, records ...types.DistrRecord) error {
+	distrInfo := k.GetDistrInfo(ctx)
+
+	deltaWeight := sdk.NewInt(0)
```

Comment:
Would be good to add comments for deltaWeight

Reply:
It's outdated